### PR TITLE
Made serializing null values optional through JsonConfiguration

### LINKED ISSUE #2650: Ignore null values in Json
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of Nancy
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/NancyFx/Nancy/issues) and [closed](https://github.com/NancyFx/Nancy/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When returning a model from a route as JSON there appears to be no option in the `Configure` method via `env.Json()` to tell the serializer not to include null values.

### Steps to Reproduce
Return an object with 2 properties set one to a value and another to be null

### System Configuration
Nancy 2.0.0-BarneyRubble

- Nancy version:
- Nancy host
  - [ ] ASP.NET
  - [ ] OWIN
  - [ ] Self-Hosted
  - [x] Other: Kestrel
- Other Nancy packages and versions:
- Environment (Operating system, version and so on): OSX, 
- .NET Framework version: dotnet core 1.1
- Additional information:

### PR DESCRIPTION
**Pull request by jchannon**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/NancyFx/Nancy/pulls) open
- [x] I have verified that I am following the Nancy [code style guidelines](https://github.com/NancyFx/Nancy/blob/45238076ad0b7f6ecabd6bae8469e30458d02efe/CONTRIBUTING.md#style-guidelines)
- [x] I have provided test coverage for my change (where applicable)

### Description
Fixes #2650

### CODE REVIEW

**Comment by khellang** [comment]
We need to remember to take this into consideration in the other serializers as well.

**Comment by jchannon** [comment]
ServiceStack already offers it. No idea about Jsonnet and protobuf however all serializers allow configuration of their settings via different config objects related to that lib.  They dont all use JsonConfiguration in Nancy

**Comment by khellang** [comment]
> They dont all use JsonConfiguration in Nancy

What? So we don't look at that configuration when creating those serializers? If I configure `JsonConfiguration`, I expect all serializers to look at that and behave the same.

**Comment by jchannon** [comment]
Correct, we never have.

https://github.com/NancyFx/Nancy.Serialization.ServiceStack#customization

https://github.com/NancyFx/Nancy.Serialization.JsonNet#customization

**Comment by khellang** [comment]
> Correct, we never have.

Maybe it's time to change that, now that we have a better configuration system?

**Comment by thecodejunkie** [comment]
👀

**Comment by thecodejunkie** [comment]
👀

File: test/Nancy.Tests.Functional/Tests/SerializerTests.cs [hunk 1]
```diff
@@ -127,6 +127,33 @@ public SerializerTests()
...
+            //When
+            var result = await browser.Post("/serializer/date", with =>
+            {
+                with.Body("{\"createdOn\":\"2016-05-06T09:00:00.0000000+02:00\"}", "application/json");
+                with.Accept("application/json");
+            });
+
+            var body = result.Body.AsString();
+
+            Assert.Equal("{\"createdOn\":\"2016-05-06T07:00:00.0000000Z\"}", body);
```
Comment (jchannon) [inline test/Nancy.Tests.Functional/Tests/SerializerTests.cs]: Nope nicked it from another test

File: src/Nancy/Json/JsonConfiguration.cs [hunk 2]
```diff
@@ -70,5 +73,10 @@ public JsonConfiguration(Encoding defaultEncoding, IList<JavaScriptConverter> co
         /// Get or sets whether enums should be treated as string
         /// </summary>
         public bool SerializeEnumToString{ get; private set; }
+
+        /// <summary>
+        /// Gets or sets if the serializer should return null values for properties on objects
+        /// </summary>
+        public bool IncludeNulls { get; set; }
```
Comment (jchannon) [inline src/Nancy/Json/JsonConfiguration.cs]: Done!

**Comment by jchannon** [comment]
GO GO GO

**Pull request merged**